editor/scene/3d/node_3d_editor_plugin.cpp

```diff
@@ -5991,23 +6235,49 @@ Node3DEditorViewport::Node3DEditorViewport(Node3DEditor *p_spatial_editor, int p
 
 	geometry.instantiate();
 
+	geometry_xray.instantiate();
+
 	ruler_line = memnew(MeshInstance3D);
 	ruler_line->set_mesh(geometry);
 	ruler_line->set_material_override(ruler_material);
 
 	ruler_line_xray = memnew(MeshInstance3D);
-	ruler_line_xray->set_mesh(geometry);
+	ruler_line_xray->set_mesh(geometry_xray);
 	ruler_line_xray->set_material_override(ruler_material_xray);
 
+	ruler_triangle_lines = memnew(MeshInstance3D);
+	Ref<ImmediateMesh> triangle_mesh = memnew(ImmediateMesh);
```

Comment:
```suggestion
	Ref<ImmediateMesh> triangle_mesh;
	triangle_mesh.instantiate();
```
Per the preferred syntax